Sources/TestingMacros/CMakeLists.txt

```diff
@@ -58,6 +58,20 @@ if(SwiftTesting_BuildMacrosAsExecutables)
   target_compile_definitions(TestingMacros PRIVATE SWT_NO_LIBRARY_MACRO_PLUGINS)
 else()
   add_library(TestingMacros SHARED)
+
+  target_link_options(TestingMacros PRIVATE "-no-toolchain-stdlib-rpath")
+  # Not setting RPATH means it requires all the dependencies are already loaded
+  # in the process, because 'plugin' directory wouldn't contain any dependencies.
+  set_property(TARGET TestingMacros PROPERTY INSTALL_RPATH)
```

Comment:
You should use `set_target_properties` to avoid calling the function twice.

Reply:
I don't know how to _unset_ a property using `set_target_properties`.

Reply:
Is it critical to change it? Or just an efficiency thing?

Reply:
Efficiency and stylistic thing I think. I believe this also work.
```cmake
set_target_properties(TestingMacros PROPERTIES
  INSTALL_RPATH ""
  BUILD_WITH_INSTALL_RPATH YES)
```

Reply:
I believe you can do both with `set_property`

```cmake
set_property(TARGET TestingMacros
    PROPERTY INSTALL_RPATH
    PROPERTY BUILD_WITH_INSTALL_RPATH YES)
```

Reply:
Hmm, are you sure? I don't think this handles multiple `PROPERTY` pairs
https://github.com/Kitware/CMake/blob/5e734dd25a466d8f6e328bf6fc49bc44a4c4c480/Source/cmSetPropertyCommand.cxx#L387